include/grpcpp/ext/otel_plugin.h

```diff
@@ -0,0 +1,99 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPCPP_EXT_OTEL_PLUGIN_H
+#define GRPCPP_EXT_OTEL_PLUGIN_H
+
+#include <grpc/support/port_platform.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <memory>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/strings/string_view.h"
+#include "opentelemetry/metrics/meter_provider.h"
+
+namespace grpc {
+
+namespace internal {
+class OpenTelemetryPluginBuilderImpl;
```

Comment:
I assume that once we have the plugin option API in place, there will no longer be any need for having two different layers here (the private API wrapped inside of the public API)?

Reply:
Yeah, I think once the experimental `CsmObservability` API goes away, we won't need to do this.

Reply:
It seems like we need the CSM plugin option implemented before we can remove the `CsmObservability` API.  And once we have the CSM plugin option implemented, the `CsmObservability` code can use that plugin option in the public API, so it won't need to reach inside the internal API anymore.  Or am I missing something here?

Reply:
I see what you mean. I wasn't originally planning on changing the implementation of `CsmObservability` to use the plugin option, but I like that idea. I will be able to remove the internal API earlier. Will do that. Thanks!